src/components/Contacts/ContactDetails/ContactDontationsTab/DonationsGraph/DonationsGraph.tsx

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable eqeqeq */
+import { Box, styled, Typography } from '@material-ui/core';
+import { DateTime } from 'luxon';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  Bar,
+  BarChart,
+  CartesianGrid,
+  Legend,
+  Tooltip,
+  XAxis,
+  YAxis,
+} from 'recharts';
+import theme from '../../../../../theme';
+import { DonationsContactFragment } from '../ContactDonationsTab.generated';
+
+const LegendText = styled(Typography)(({ theme }) => ({
+  margin: theme.spacing(3, 0),
+  writingMode: 'vertical-rl',
+  textOrientation: 'mixed',
+}));
+
+const GraphContainer = styled(Box)(({ theme }) => ({
+  display: 'flex',
+  margin: theme.spacing(0, 2, 0, 0),
+}));
+
+const NoDonationsBox = styled(Box)(({ theme }) => ({
+  margin: theme.spacing(4),
+}));
+interface DonationsGraphProps {
+  donations: DonationsContactFragment | null;
+}
+
+export const DonationsGraph: React.FC<DonationsGraphProps> = ({
+  donations,
+}) => {
+  const { t } = useTranslation();
+  const donationsForThisYear = donations?.nodes.filter((donation) => {
+    return (
+      DateTime.fromISO(donation.donationDate).toJSDate() >
+      DateTime.local().minus({ year: 1 }).toJSDate()
+    );
+  });
+
+  const donationsForLastYear = donations?.nodes.filter((donation) => {
+    return (
+      DateTime.fromISO(donation.donationDate).toJSDate() <
+        DateTime.local().minus({ year: 1 }).toJSDate() ||
+      DateTime.fromISO(donation.donationDate).toJSDate() >
+        DateTime.local().minus({ year: 2 }).toJSDate()
+    );
+  });
+
+  let convertedCurrency = '';
```

Comment:
Looks like you are just pulling out the last donation in the array from either `donationsForThisYear` or `donationsForLastYear` and looking up the currency. Maybe that doesn't need to be in the loop since you just throw all the intermediate values away. Maybe there's a better way to look this up for a contact...